traversal/planner/PlannerEdge.java

```diff
@@ -232,26 +236,29 @@ private void recordCost() {
         protected void updateOptimiserCoefficients() {
             assert costLastRecorded == safeCost();
             planner.optimiser().setObjectiveCoefficient(varIsMinimal, log(1 + safeCost()));
-            initialiseMinimalEdgeConstraint();
         }
 
-        public void initialiseOptimiserValues() {
-            setInitialSelected();
-            setInitialMinimal();
+        protected void updateOptimiserConstraints() {
+            updateMinimalEdgeConstraint();
+        }
+
+        public void setOptimiserValues() {
+            setSelected();
+            setMinimal();
             isInitialised = true;
         }
 
-        public void setInitialSelected() {
-            varIsSelected.setInitial(from().varOrderNumber.initial() < to().varOrderNumber.initial());
+        public void setSelected() {
+            varIsSelected.setValue(from().getOrder() < to().getOrder());
         }
 
-        public void setInitialMinimal() {
-            assert varIsSelected.hasInitial();
-            varIsMinimal.setInitial(
-                    varIsSelected.initial() &&
+        public void setMinimal() {
+            assert varIsSelected.hasValue();
+            varIsMinimal.setValue(
+                    varIsSelected.value() &&
```

Comment:
looks great now! I'm pleased with the current state now - consistent, accurate, and minimal!